gcloud/datastore/query.py

```diff
@@ -347,10 +342,10 @@ class Iterator(object):
         datastore_pb.QueryResultBatch.MORE_RESULTS_AFTER_LIMIT,
     )
 
-    def __init__(self, query, connection, limit=None, offset=0,
+    def __init__(self, query, client, limit=None, offset=0,
```

Comment:
Also, if the `_client` is bound to the query, shouldn't `client` be optional? I suppose it's moot as `Iterator` is effectively a non-public class.